### ISSUE TITLE
[Bug]: Module Conflict with Hidden Initiative

### ISSUE THREAD

**Issue by Cubius** [arcanistzed/ctg#13]
### Expected Behavior

Initiative should progress in descending order.

### Current Behavior

[Hidden Initiative](https://foundryvtt.com/packages/hidden-initiative) will generate a random order of combatants to display initially just for players - not for GMs - and with each turn reorder to reveal the actual order.

As of the latest version of Combat Tracker Groups, when starting a combat, when skipping to the next turn will not follow the numbers of initiative in descending order, but rather jump through the combatants in a random sequence instead. I suspect that it is following the fake order Hidden Initiative is generating instead.

### Steps to Reproduce

1. Install both [Hidden Initiative](https://foundryvtt.com/packages/hidden-initiative) and [Combat Tracker Groups](https://foundryvtt.com/packages/ctg).
2. Start a combat. More combatants is better to illustrate the issue, I recommend at least 5 combatants.
3. Roll initiative and click to advance to the next turn.

### Version

1.1.0

### Foundry VTT Version

9.269

### Operating System

Linux

### Browser / App

Chrome, Firefox

### Game System

dnd5e (1.6.3)

### Modules Disabled

- [X] I can confirm that disabling all of my other modules doesn't fix this issue.

**Comment by arcanistzed** [comment]
Could you show me a screenshot of your Combat Tracker Groups module settings? Specifically, I'm curious if you have "Sort Combatants" enabled and which Grouping Modes you are using.

**Comment by Cubius** [comment]
Sort Combatants is indeed enabled as you can see below:

![grafik](https://user-images.githubusercontent.com/15717384/176995343-3577986b-8a4e-4c40-8970-f9d1f349f790.png)

![grafik](https://user-images.githubusercontent.com/15717384/176995350-33fe61e3-cbd3-4117-b154-525801b61b00.png)

I first noticed this issue in a combat with a group of enemies, but it also occurs with ungrouped individual actors.

**Comment by arcanistzed** [comment]
Which of those grouping modes are you actually using?

**Comment by arcanistzed** [comment]
With the default "Initiative" mode, the turn order advances from highest to lowest initiative properly for me. If you disable "Group Skipping" it's even easier to see that it is working properly.

**Comment by Cubius** [comment]
I think I see what you are talking about. It seems I did not fully understand the grouping functionality.

I would rephrase my issue to be then that the various modes do not just group combatants, but sort them too.

'Grouping' by name just sorts individual actors by name instead of initiative.

![grafik](https://user-images.githubusercontent.com/15717384/177014580-cc883c7e-7655-4b08-9e18-a065593534d3.png)

The same seems to apply to the other modes. Grouping by 'None' meanwhile appears to just be a seemingly random order - I would suspect the order combatants were added to the tracker.

![grafik](https://user-images.githubusercontent.com/15717384/177014546-b21dc1bf-e6f3-4ca1-9a2e-8ade8eddac6d.png)

If this is functioning as intended, obviously mark this issue as closed, but I would expect combatants to be grouped by the desired metric, while combat still progresses in descending order through initiative rather than completely disregarding initiative and going alphabetically or otherwise.

**Comment by arcanistzed** [comment]
I think that what you are describing can be achieved by simply disable the "Sort Combatants" setting. The goal of that feature is to sort the combatants by the same attribute that they are being grouped by which makes "Group Skipping" more predictable.

**Comment by Cubius** [comment]
Apparently turning off "Sort Combatants" does what it's name implies.

A little counter-intuitive to the description of that setting.

I saw a fair bit of the tracker jumping backwards or randomly skipping around with the setting still enabled. Potentially that might have been an issue with having two accounts with GM permissions on different sorting modes. I'll investigate if that has caused some of my issues, but for now close the issue.

Thanks for the help though!

**Comment by arcanistzed** [comment]
Yeah. The "Sort Combatants" setting basically decides whether CTG applies any of its own logic to the turn order, so disabling it will mean that you are using the default sorting of your game system.
